### PR TITLE
Add empty gitignore during source distribution builds

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

It's not clear _why_ it matters in this case, but Hatchling respecting the `.gitignore` in our cache is causing certain builds to fail. This PR adds an empty `.gitignore` to the `built-wheels-v0` bucket and ensures that all unzipped distributions are stored in temporary folders within that bucket.

## Test Plan

Run `cargo run pip install "local_not_used_with_sdist_a @ file:///Users/crmarsh/Downloads/local_not_used_with_sdist_a-1.2.3" --reinstall --verbose --verbose -n`.

Then:

```
❯ ls .venv/lib/python3.12/site-packages/
__pycache__                                 _virtualenv.pth                             _virtualenv.py                              local_not_used_with_sdist_a                 local_not_used_with_sdist_a-1.2.3.dist-info
```

### CODE REVIEW

**Comment by zanieb** [comment]
See https://github.com/pypa/hatch/issues/1273 for details on the upstream issue — tldr if a source distribution does not have a `.gitignore` file hatchling will traverse upward until it finds one — which in our case is our cache's `.gitignore` file which has a `*` in it.

We previously added a `.git` directory to prevent upward traversal, but `hatch` does yet support stopping traversal at git project boundaries.

File: crates/uv-cache/src/lib.rs [hunk 1]
```diff
@@ -257,6 +257,20 @@ impl Cache {
             .write(true)
             .open(root.join(".git"))?;
 
+        // Add an empty .gitignore to the build bucket, to ensure that the cache's own .gitignore
+        // doesn't interfere with source distribution builds. (Build backends like hatchling will
+        // traverse upwards to look for .gitignore files.)
```
Comment (zanieb) [inline crates/uv-cache/src/lib.rs:262]: I'd rephrase as "Build backends (like hatchling) may traverse upwards..." but no strong feelings.

**Review by zanieb — APPROVED** [review]

**Comment by charliermarsh** [comment]
Hah, that test now _correctly_ fails.

**Comment by charliermarsh** [comment]
Need https://github.com/astral-sh/uv/pull/2462 to merge first.

**Comment by charliermarsh** [comment]
Ah shoot, this was accidentally merged as part of https://github.com/astral-sh/uv/pull/2462.
